dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandler.kt

```diff
@@ -53,30 +56,40 @@ internal class DatadogLogHandler(
         tags: Set<String>,
         timestamp: Long
     ): Log {
-
-        var traceId: String? = null
-        var spanId: String? = null
+        val combinedAttributes = attributes.toMutableMap()
         if (bundleWithTraces && GlobalTracer.isRegistered()) {
             val tracer = GlobalTracer.get()
-            traceId = tracer.traceId()
-            spanId = tracer.spanId()
+            val activeContext = tracer.activeSpan()?.context()
+            if (activeContext != null) {
+                combinedAttributes[TAG_TRACE_ID] = activeContext.toTraceId()
+                combinedAttributes[TAG_SPAN_ID] = activeContext.toSpanId()
+            }
+        }
+        if (bundleWithRum && GlobalRum.isRegistered()) {
+            val activeContext = GlobalRum.getRumContext()
+            combinedAttributes[TAG_APPLICATION_ID] = activeContext.applicationId.toString()
+            combinedAttributes[TAG_SESSION_ID] = activeContext.sessionId.toString()
+            combinedAttributes[TAG_VIEW_ID] = activeContext.viewId?.toString()
```

Comment:
I think all this keys (`TAG_APPLICATION_ID`, `TAG_SESSION_ID`, `TAG_VIEW_ID`) should also be added to `reservedAttributes` in `LogSerializer`, otherwise user might override them unconsciously. This might IMO happen, as not all keys are namespaced with `dd`, i.e. `view.id` is quite generic.

Reply:
The rule for reserved attributes was to only keep the sensitive ones; then if users override those, that's their problem.